src/components/Comments.jsx

```diff
@@ -1,54 +1,114 @@
-            setCommentsByArticle(comments)
-            setIsLoading(false)
-            setIsError(false)
-        }).catch((err) => {
-            setIsLoading(false)
-            setIsError(true)
-          });
-    }, [article_id])
+  const { article_id } = useParams();
+  const [commentsByArticle, setCommentsByArticle] = useState([]);
+  const [postComment, setPostComment] = useState({ username: "grumpy19" });
+  const [commentAdd, setCommentAdd] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+  const dateConfig = {
+    day: "numeric",
+    month: "short",
+    year: "numeric",
+  };
+  const timeConfig = {
+    hour: "2-digit",
+    minute: "2-digit",
+  };
+  useEffect(() => {
+    setIsLoading(true);
+    getCommentsByArticleId(article_id)
+      .then((comments) => {
+        setCommentsByArticle(comments);
+        setIsLoading(false);
+        setIsError(false);
+      })
+      .catch((err) => {
+        setIsLoading(false);
+        setIsError(true);
+      });
+  }, [article_id]);
 
-    if(isLoading){
-        return (
-            <p>Loading Comments...</p>
-        )
+  function commentInput(e) {
+    const commentBtn = document.getElementById("comment-btn");
+    if (e.target.value !== "") {
+      commentBtn.disabled = false;
+      setPostComment({ ...postComment, body: e.target.value });
+    } else {
+      commentBtn.disabled = true;
     }
-    if(isError){
-        return (
-            <p>Loading Comments...</p>
-        )
-    }
-    return (
-        <section>
-            <h3>Comments</h3>
-            <ul>
-                {commentsByArticle.map((comment,i) => {
-                    return (
-                        <li key={i}>
-                            <section>
-                                <h4>{comment.author}</h4>
-                                <time dateTime="">{comment.created_at}</time>
-                                <p>{comment.body}</p>
-                                <span>{comment.votes}</span>
-                            </section>
-                        </li>
-                    )
-                })}
-            </ul>
-        </section>
-    );
+  }
+
+  function commentSubmit(e) {
+    e.preventDefault();
+    postCommentByArticleId(article_id, postComment)
+      .then((comment) => {
+        setCommentAdd(true);
+        setCommentsByArticle([comment, ...commentsByArticle]);
+      })
+      .catch((err) => {
+        console.log(err);
+      });
+  }
+
+  if (isLoading) {
+    return <p>Loading Comments...</p>;
+  }
+  if (isError) {
+    return <p>Loading Comments...</p>;
+  }
+  return (
+    <section>
+      <h3>Comments</h3>
+      {!commentAdd ? (
+        <form onSubmit={(event) => commentSubmit(event)}>
+          <input
+            type="text"
+            name="comment"
+            placeholder="Enter a comment"
+            onInput={(event) => commentInput(event)}
+          />
```

Comment:
This is not an example of a controlled input. I would recommend revising the notes on this:
l2c.northcoders.com/courses/sd-notes/front-end/#sectionId=controlled-components,step=controlled-input